source/geometries/JaszczakPhantom.cc

```diff
@@ -0,0 +1,200 @@
+{
+  auto cyl_name = "CYLINDER";
+  auto cylinder_solid =
+    new G4Tubs(cyl_name, 0, (cylinder_inner_diam_+2*cylinder_thickn_)/2,
+               (cylinder_height_+2*cylinder_thickn_)/2., 0, twopi);
+
+  G4Material* aluminum = G4NistManager::Instance()->FindOrBuildMaterial("G4_Al");
+  auto cylinder_logic =
+    new G4LogicalVolume(cylinder_solid, aluminum, cyl_name);
+
+  this->SetLogicalVolume(cylinder_logic);
+
+
+  auto water_name = "WATER_BCKG";
+  auto water_solid =
+    new G4Tubs(water_name, 0, cylinder_inner_diam_/2, cylinder_height_/2., 0, twopi);
+
+  G4Material* water = G4NistManager::Instance()->FindOrBuildMaterial("G4_WATER");
+  auto water_logic =
+    new G4LogicalVolume(water_solid, water, water_name);
+  G4VPhysicalVolume* water_phys =
+    new G4PVPlacement(0, G4ThreeVector(0., 0., 0.), water_logic, water_name, cylinder_logic,
+                      false, 0, true);
+
+  cyl_gen_ = new CylinderPointSampler2020(water_phys);
+
+  // Spheres
+  std::vector<G4double> sphere_radii =
+    {sphere1_d_/2, sphere2_d_/2, sphere3_d_/2, sphere4_d_/2, sphere5_d_/2, sphere6_d_/2};
+  auto radius_pos = cylinder_inner_diam_/4.;
+  auto z_pos      = - cylinder_height_/2. + sphere_height_;
+
+  for (unsigned long i=0; i<sphere_radii.size(); i++) {
+    BuildSpheres(i, sphere_radii[i], radius_pos, z_pos, water_logic, water);
+  }
+
+  // Rods
+  std::vector<G4double> rod_radii =
+    {rod1_d_/2, rod2_d_/2, rod3_d_/2, rod4_d_/2, rod5_d_/2, rod6_d_/2};
+  z_pos = - cylinder_height_/2. + rod_height_/2;
+  for (unsigned long i=0; i<rod_radii.size(); i++) {
+    BuildRods(i, rod_radii[i], z_pos, water_logic, water);
+  }
+
+  // Relative actvities
+  auto max_activity = std::max(sphere_activity_, std::max(bckg_activity_, rod_activity_));
+  bckg_activity_   /= max_activity;
+  sphere_activity_ /= max_activity;
+  rod_activity_    /= max_activity;
+
+  G4cout << "*** Relative activities (background, spheres, rods) ***" << G4endl;
+  G4cout << bckg_activity_ << ", " << sphere_activity_  << ", " << rod_activity_ << G4endl;
+
+}
+
+
+void JaszczakPhantom::BuildSpheres(unsigned long n, G4double r, G4double r_pos, G4double z_pos,
+                                   G4LogicalVolume* mother_logic, G4Material* mat) const
+{
+  auto sphere_name = "SPHERE" + std::to_string(n);
+  auto sphere_solid = new G4Orb(sphere_name, r);
+  auto sphere_logic = new G4LogicalVolume(sphere_solid, mat, sphere_name);
+  auto angle = n * 60 * deg;
+  auto x_pos = r_pos * cos(angle);
+  auto y_pos = r_pos * sin(angle);
+  new G4PVPlacement(0, G4ThreeVector(x_pos, y_pos, z_pos), sphere_logic,
+                    sphere_name, mother_logic, false, 0, true);
+}
+
+
+
+void JaszczakPhantom::BuildRods(unsigned long n, G4double r, G4double z_pos,
+                                G4LogicalVolume* mother_logic, G4Material* mat) const
+  {
+    auto diam = 2 * r;
+
+    // Sector displacement from centre, to accommodate gap between sectors
+    auto gap =  14.4 * mm;
```

Comment:
Is this `gap` standard or you computed the distance to fit all rods?

Reply:
I have translated Jacek's code to our simulations, I have assumed it was correct.